crates/cairo-profiler/src/trace_reader.rs

```diff
@@ -168,6 +179,77 @@ fn collect_samples<'a>(
             sierra_gas_tracking,
         );
 
+        let mut trigger_idx = 0;
+
+        while trigger_idx < function_level_profiling_info.nested_call_triggers.len() {
+            let trigger = &function_level_profiling_info.nested_call_triggers[trigger_idx];
+            let traced_syscall = &trigger.last().unwrap().function_name().0;
+
+            let Some(&sub_trace) = entrypoint_calls.peek() else {
+                // There are syscall triggers, but no entrypoints left
+                // It is acceptable only for Deploy syscalls, as we filtered out snforge's DeployWithoutConstructor-s
+                if traced_syscall == "Deploy" {
+                    trigger_idx += 1;
+                    continue; // just skip this Deploy trigger and move on
+                }
+                ui::err(format!(
+                    "Found syscall {traced_syscall} in the program trace, that do not have corresponding calls in trace file!"
+                ));
+                panic!("Too few EntryPointCalls for triggers");
+            };
+
+            let expected_syscall = map_entrypoint_to_syscall(&sub_trace.entry_point);
+
+            if traced_syscall == expected_syscall {
+                entrypoint_calls.next();
+                trigger_idx += 1;
+
+                let mut triggered_call_stack = current_entrypoint_call_stack.clone();
+                triggered_call_stack.extend(trigger.clone());
+
+                children_resources.add_resources(collect_samples(
+                    samples,
+                    &mut triggered_call_stack,
+                    sub_trace,
+                    compiled_artifacts_cache,
+                    profiler_config,
+                    versioned_constants,
+                    sierra_gas_tracking,
+                )?);
+            } else if expected_syscall == "Deploy" {
+                // snforge can sometimes insert a Deploy nested_call that is not a syscall!
+                ui::warn("No 'Deploy' syscall in trace but CONSTRUCTOR nested call found!");
+                entrypoint_calls.next();
+                children_resources.add_resources(collect_samples(
+                    samples,
+                    current_entrypoint_call_stack,
+                    sub_trace,
+                    compiled_artifacts_cache,
+                    profiler_config,
+                    versioned_constants,
+                    sierra_gas_tracking,
+                )?);
+            } else if traced_syscall == "Deploy" {
+                // keep looking for matching nested_call
+                trigger_idx += 1;
```

Comment:
the plan is to fix this in forge, but the decision will still have to be made if we want to leave this "workaround" here for backwards compatibility with older forge versions (I'd say yes - at least for now)